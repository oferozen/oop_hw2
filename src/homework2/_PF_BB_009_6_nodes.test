#syntax:
# CreateNode n w
# CreateGraph n
# AddNode g n
# AddEdge A n3 n1 
# FindPath g n1 n2 -> n3 n4

CreateGraph g
CreateNode n1 30
CreateNode n2 10
CreateNode n3 10
CreateNode n4 1
CreateNode n5 1
CreateNode n6 20

AddNode g n1
AddNode g n2
AddNode g n3
AddNode g n4
AddNode g n5
AddNode g n6

# 2 paths:
# (1) n4 -> n2 -> n1 -> n5
# (2) n4 -> n3 -> n6 -> n5

# path (1)
AddEdge g n4 n2
AddEdge g n2 n1
AddEdge g n1 n5

# path (2)
AddEdge g n4 n3
AddEdge g n3 n6
AddEdge g n6 n5

# shortest path should be (2)

FindPath g n4 -> n5
FindPath g n5 -> n4

# Adding path (3) n4 -> n7 -> n5
CreateNode n7 25
AddNode g n7
AddEdge g n4 n7
AddEdge g n7 n5
FindPath g n4 -> n5
