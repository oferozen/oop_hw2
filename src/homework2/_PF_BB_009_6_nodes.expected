#syntax:
# CreateNode n w
# CreateGraph n
# AddNode g n
# AddEdge A n3 n1 
# FindPath g n1 n2 -> n3 n4

created graph g
created node n1 with cost 30
created node n2 with cost 10
created node n3 with cost 10
created node n4 with cost 1
created node n5 with cost 1
created node n6 with cost 20

added node n1 to g
added node n2 to g
added node n3 to g
added node n4 to g
added node n5 to g
added node n6 to g

# 2 paths:
# (1) n4 -> n2 -> n1 -> n5
# (2) n4 -> n3 -> n6 -> n5

# path (1)
added edge from n4 to n2 in g
added edge from n2 to n1 in g
added edge from n1 to n5 in g

# path (2)
added edge from n4 to n3 in g
added edge from n3 to n6 in g
added edge from n6 to n5 in g

# shortest path should be (2)

shortest path in g: n4 n3 n6 n5
shortest path in g:

# Adding path (3) n4 -> n7 -> n5
created node n7 with cost 25
added node n7 to g
added edge from n4 to n7 in g
added edge from n7 to n5 in g
shortest path in g: n4 n7 n5
